### PR TITLE
CORE-7629: Use resultList instead of resultStream when querying for CPI metadata

### PR DESCRIPTION
**Pull request by fowlerrr**

### CODE REVIEW

File: processors/db-processor/src/main/kotlin/net/corda/processors/db/internal/reconcile/db/DbReconcilerReader.kt [hunk 1]
```diff
@@ -97,12 +96,14 @@ class DbReconcilerReader<K : Any, V : Any>(
...
+    override fun getAllVersionedRecords(): List<VersionedRecord<K, V>>? {
         return try {
             val em = entityManagerFactory!!.createEntityManager()
             val currentTransaction = em.transaction
-            currentTransaction.begin()
-            doGetAllVersionedRecords(em).onClose {
+            return try {
+                currentTransaction.begin()
+                doGetAllVersionedRecords(em)
+            } finally {
```
Comment (fowlerrr) [inline processors/db-processor/src/main/kotlin/net/corda/processors/db/internal/reconcile/db/DbReconcilerReader.kt:106]: These nested try's are weird.  But I couldn't think of another way to catch all of the issues _including_ a throw when creating the `EntityManager`.  Might be my brain is just tired...

File: libs/virtual-node/cpi-datamodel/src/main/kotlin/net/corda/libs/cpi/datamodel/CpiMetadataEntity.kt [hunk 2]
```diff
@@ -136,12 +135,12 @@ data class CpiMetadataEntityKey(
...
     // Joining the other tables to ensure all data is fetched eagerly
     return createQuery(
         "FROM ${CpiMetadataEntity::class.simpleName} cpi_ " +
                 "INNER JOIN FETCH cpi_.cpks cpk_ " +
                 "INNER JOIN FETCH cpk_.metadata cpk_meta_",
         CpiMetadataEntity::class.java
-    ).resultStream
-}
\ No newline at end of file
+    ).resultList
```
Comment (fowlerrr) [inline libs/virtual-node/cpi-datamodel/src/main/kotlin/net/corda/libs/cpi/datamodel/CpiMetadataEntity.kt:145]: This here is the crux of the change.  The streams seem to be failing us.

File: components/reconciliation/reconciliation/src/main/kotlin/net/corda/reconciliation/ReconcilerReader.kt [hunk 3]
```diff
@@ -22,7 +21,7 @@ interface ReconcilerReader<K : Any, V : Any> {
      * Subsequently, the calling service, that should be following [ReconcilerReader], will as well get
      * notified of the [ReconcilerReader]'s changed status by a RegistrationStatusChangeEvent.
      */
-    fun getAllVersionedRecords(): Stream<VersionedRecord<K, V>>?
+    fun getAllVersionedRecords(): List<VersionedRecord<K, V>>?
```
Comment (driessamyn) [inline components/reconciliation/reconciliation/src/main/kotlin/net/corda/reconciliation/ReconcilerReader.kt:24]: We shouldn't change the API. It's a stream for a good reason, as Lists will not be sustainable in the long term.

**Review by driessamyn — CHANGES_REQUESTED** [review]

File: libs/virtual-node/cpi-datamodel/src/main/kotlin/net/corda/libs/cpi/datamodel/CpiMetadataEntity.kt (hunk 2 above)
Comment (driessamyn) [reply libs/virtual-node/cpi-datamodel/src/main/kotlin/net/corda/libs/cpi/datamodel/CpiMetadataEntity.kt:145]: Have you tried to log the generated SQL query and inspected it?
I wonder if it's an order problem. Maybe the CPKs come in out of order.

Comment (fowlerrr) [reply libs/virtual-node/cpi-datamodel/src/main/kotlin/net/corda/libs/cpi/datamodel/CpiMetadataEntity.kt:145]: There's nothing wrong with the SQL.  It works fine.  It's something about the stream (or the stream.toList()) that's the problem.

Getting a direct list works fine.  As I understand it (though I'll just quickly verify it myself) a stream works okay.  It's the conversion of the stream to the list that's breaking.

**Comment by fowlerrr** [comment]
We're backing out 94.3% of these changes so I'll raise a new PR

**Comment by Tom-Fitzpatrick** [comment]
Sorry, I was looking at this PR in IntelliJ and accidentally closed it

**Pull request closed without merging**